src/main/java/org/broadinstitute/consent/http/authentication/AuthorizationHelper.java

```diff
@@ -0,0 +1,104 @@
+package org.broadinstitute.consent.http.authentication;
+
+import com.google.common.cache.Cache;
+import com.google.inject.Inject;
+import jakarta.ws.rs.NotFoundException;
+import jakarta.ws.rs.WebApplicationException;
+import java.util.Map;
+import org.broadinstitute.consent.http.filters.ClaimsCache;
+import org.broadinstitute.consent.http.models.AuthUser;
+import org.broadinstitute.consent.http.models.User;
+import org.broadinstitute.consent.http.models.sam.UserStatus;
+import org.broadinstitute.consent.http.models.sam.UserStatusInfo;
+import org.broadinstitute.consent.http.service.UserService;
+import org.broadinstitute.consent.http.service.sam.SamService;
+import org.broadinstitute.consent.http.util.ConsentLogger;
+
+/**
+ * Helper class for authorization and authentication. This class contains methods to build AuthUser
+ * objects from request headers, retrieve user status information from Sam, and check user roles.
+ */
+public class AuthorizationHelper implements ConsentLogger {
+
+  protected final ClaimsCache claimsCache;
+  protected final SamService samService;
+  protected final UserService userService;
+
+  @Inject
+  public AuthorizationHelper(SamService samService,
+      UserService userService) {
+    this.claimsCache = ClaimsCache.getInstance();
+    this.samService = samService;
+    this.userService = userService;
+  }
+
+  protected Cache<String, Map<String, String>> getCache() {
+    return claimsCache.cache;
+  }
+
+  protected AuthUser buildAuthUserFromHeaders(Map<String, String> headers) {
+    String aud = headers.get(ClaimsCache.OAUTH2_CLAIM_aud);
+    String token = headers.get(ClaimsCache.OAUTH2_CLAIM_access_token);
+    String email = headers.get(ClaimsCache.OAUTH2_CLAIM_email);
+    String name = headers.get(ClaimsCache.OAUTH2_CLAIM_name);
+    // Name is not a guaranteed header
+    if (name == null || name.equals("unknown")) {
```

Comment:
Given the name is not a guaranteed header, maybe we shouldn't even take the value if one is provided - because it doesn't seem like we rely on it.

Reply:
Name is guaranteed for Google authentications and only not guaranteed for MS authentications. Google is the majority of our authentications and this does give us valuable info to improved the initial UX.